device_doctor/lib/src/device.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'config.dart';
+import 'ios_device.dart';
+
+/// The root of the API for controlling devices.
+DeviceDiscovery get devices => DeviceDiscovery();
+
+/// Operating system on the devices that this host is configured to test.
+enum DeviceOperatingSystem { android, ios }
+
+/// Discovers available devices and chooses one to work with.
+abstract class DeviceDiscovery {
+  factory DeviceDiscovery() {
+    switch (config.deviceOperatingSystem) {
+      case DeviceOperatingSystem.ios:
+        return IosDeviceDiscovery();
+      default:
+        throw StateError('Unsupported device operating system: {config.deviceOperatingSystem}');
```

Comment:
nit: is this missing $? ${config.deviceOperatingSystem}

Reply:
Good catch! Updated.